### PR TITLE
Mac manual build procedure update

### PR DESCRIPTION
**Pull request by nobugshere**
This relates to the discussion in issue #949 and PR #484.

Here are some general updates to the manual procedures and scripts for macOS.  The intent is to align the manual procedures with the current GitHub release workflow.
- README.MAC.DEVELOPERS.md - Describe the current requirements and the default universal build procedures.
- osx-build-wx, macos-latest.yml - Update for the current wxWidgets version and to retrieve the default parameters from the workflow file.  Some older code was retained for reference with comments indicating it's no longer supported. Added -a and -w options to override the defaults. The -t option was used twice, so I changed one to -T.
- create-dmg, Makefile - Minor updates to default to a universal build.
- version.wx - Since the default build has changed so much, I took the liberty of bumping the version number to 1.16.01, assuming a future release will be 1.17. In retrospect, this will apply to the Linux build as well.

The development and testing was on an M1 Mac running macOS 13.3.1 Ventura.  The build product was also tested on an M1 Monterey VM.

### CODE REVIEW

File: Misc/osx-build-wx [hunk 1]
```diff
@@ -83,65 +125,144 @@ while getopts "dc:t:nk:t:" opt; do
...
-            else
-				error Invalid directory passed for toolchain dir: $OPTARG
-            fi
-            ;;
+		k)
+			if [ -d $OPTARG ]
+			then
+				SDKPATH=$OPTARG
+			else
+				error "Invalid directory passed for SDK path: $OPTARG"
```
Comment (dskrvk) [inline Misc/osx-build-wx:136]: Should other error messages above be similarly quoted?

File: Misc/osx-build-wx [hunk 2]
```diff
@@ -83,65 +125,144 @@ while getopts "dc:t:nk:t:" opt; do
...
+        fi
+        # This is actually where 'make install' will put things.
+        WX_BUILD_DIR=$WX_PREFIX
+
+        # Fetch some parameters from the macos-latest.yaml workflow
+        if [ -z $WX_ARCH ]
+        then
+            eval `grep WX_ARCH= $WORKFLOW | sed -E -e 's/^.*"(.*)".*$/\1/'`
+        fi
+        eval `grep MACOS_VER_MIN= $WORKFLOW | sed -E -e 's/^.*"(.*)".*$/\1/'`
```
Comment (dskrvk) [inline Misc/osx-build-wx:185]: Am I missing something or is the regex the same as on the previous line?

File: Misc/osx-build-wx [hunk 3]
```diff
@@ -31,13 +39,35 @@ SDKPATH=
 # This is where gcc, g++ and ld required to build with 10.4 SDK should be
 TOOLCHAIN_DIR=
 
+# "make install" base directory.  For safety, we set it to pwd and don't recommend installing
```
[the remark this answers is not in the document]
Comment (nobugshere) [reply Misc/osx-build-wx:42]: Fixed, thanks.

File: Misc/osx-build-wx (hunk 1 above)
Comment (nobugshere) [reply Misc/osx-build-wx:136]: Fair point.  Fixed.

File: Misc/osx-build-wx (hunk 2 above)
Comment (nobugshere) [reply Misc/osx-build-wx:185]: The regex is the same but not the values being extracted (WX_ARCH vs. MACOS_VER_MIN).

File: Misc/osx-build-wx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dskrvk) [reply Misc/osx-build-wx:42]: I mean line 42. Still looks cut off.

[the remark this answers is not in the document]
Comment (nobugshere) [reply Misc/osx-build-wx]: Oops.  Thank you.

**Comment by ronys** [comment]
Much appreciated!

**Pull request merged**